### PR TITLE
feat(material/schematics/navigation): enable routing option

### PR DESCRIPTION
**Pull request by santoshyadav198613**
~Please do not review as of now~

### CODE REVIEW

**Comment by Splaktar** [comment]
Relates to PR https://github.com/angular/components/pull/19341.

File: src/material/schematics/ng-generate/navigation/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.html.template [hunk 1]
```diff
@@ -5,9 +5,9 @@
       [opened]="(isHandset$ | async) === false">
     <mat-toolbar>Menu</mat-toolbar>
     <mat-nav-list>
-      <a mat-list-item href="#">Link 1</a>
-      <a mat-list-item href="#">Link 2</a>
-      <a mat-list-item href="#">Link 3</a>
+      <a mat-list-item <% if(routing) routerLink="/" %> <% if(!routing) href="#" %> >Link 1</a>
+      <a mat-list-item routerLink="/">Link 2</a>
```
Comment (crisbeto) [inline src/material/schematics/ng-generate/navigation/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.html.template:9]: Don't we need `if(routing)` checks on these two links as well? Also you might be able to use a ternary here so it's all in the same expression.

File: src/material/schematics/ng-generate/navigation/index.spec.ts [hunk 2]
```diff
@@ -155,4 +155,17 @@ describe('material-navigation-schematic', () => {
...
     });
   });
+  describe('router option', () => {
+    it('should respect the option value', async () => {
+      const tree =
+        await runner
+          .runSchematicAsync(
+            'navigation', { routing: true, ...baseOptions }, await createTestApp(runner))
+          .toPromise();
+      console.log(tree.files);
```
Comment (crisbeto) [inline src/material/schematics/ng-generate/navigation/index.spec.ts:165]: Leftover `console.log`.

File: src/material/schematics/ng-generate/navigation/schema.json [hunk 3]
```diff
@@ -92,6 +92,11 @@
       "type": "boolean",
       "default": false,
       "description": "Specifies if the component is an entry component of declaring module."
+    },
+    "routing": {
+      "type": "boolean",
+      "default": false,
+      "description": "Specifies if Angular routing is enabled."
```
Comment (crisbeto) [inline src/material/schematics/ng-generate/navigation/schema.json:99]: "Specifies if..." -> "Specifies whether..."

File: src/material/schematics/ng-generate/navigation/index.spec.ts (hunk 2 above)
Comment (santoshyadav198613) [reply src/material/schematics/ng-generate/navigation/index.spec.ts]: Removed.

File: src/material/schematics/ng-generate/navigation/schema.json (hunk 3 above)
Comment (santoshyadav198613) [reply src/material/schematics/ng-generate/navigation/schema.json]: Done.

File: src/material/schematics/ng-generate/navigation/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.html.template (hunk 1 above)
Comment (santoshyadav198613) [reply src/material/schematics/ng-generate/navigation/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.html.template]: Done.

**Comment by santoshyadav198613** [comment]
Hi @Splaktar ,
This is ready for review now.

File: src/material/schematics/ng-generate/navigation/schema.json [hunk 4]
```diff
@@ -92,6 +92,11 @@
       "type": "boolean",
       "default": false,
       "description": "Specifies if the component is an entry component of declaring module."
+    },
+    "routing": {
```
Comment (alan-agius4) [inline src/material/schematics/ng-generate/navigation/schema.json:96]: I personally think that the schematic should be doing this check based on the app module and not provided as an option.

Comment (devversion) [reply src/material/schematics/ng-generate/navigation/schema.json:96]: That might be overly complicated as we'd need to check the imports of the module. These checks are not reliable most of the time either as in reality the Angular compiler supports partial evaluation (i.e. module can be specified in a separate file through an exported variable).

Surely we could use the static interpreter as in FW update migrations, but as said, that might be out of scope.

File: src/material/schematics/ng-generate/navigation/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.html.template [hunk 5]
```diff
@@ -5,9 +5,9 @@
       [opened]="(isHandset$ | async) === false">
     <mat-toolbar>Menu</mat-toolbar>
     <mat-nav-list>
-      <a mat-list-item href="#">Link 1</a>
-      <a mat-list-item href="#">Link 2</a>
-      <a mat-list-item href="#">Link 3</a>
+      <a mat-list-item <%= routing === true ? 'routerLink="/"' : 'href="#"' %>>Link 1</a>
```
Comment (crisbeto) [inline src/material/schematics/ng-generate/navigation/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.html.template:8]: You should be able to do just `routing ? 'routerLink="/"' : 'href="#"'`.

Comment (santoshyadav198613) [reply src/material/schematics/ng-generate/navigation/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.html.template]: @crisbeto done.

File: src/material/schematics/ng-generate/navigation/schema.json [hunk 6]
```diff
@@ -92,6 +92,11 @@
...
+      "type": "boolean",
+      "default": false,
+      "description": "Specifies whether Angular routing is enabled."
```
Comment (jelbourn) [inline src/material/schematics/ng-generate/navigation/schema.json:99]: I would add just a tiny bit more to this description:
"This controls whether anchor elements use `href` or `routerLink`"

Comment (santoshyadav198613) [reply src/material/schematics/ng-generate/navigation/schema.json]: Done @jelbourn .

**Comment by santoshyadav198613** [comment]
Hi @jelbourn ,
Do I need to update commit message?

**Comment by jelbourn** [comment]
It would help if it was just  `feat(material/schematics)`, but we can also change it when we merge

**Comment by santoshyadav198613** [comment]
> It would help if it was just `feat(material/schematics)`, but we can also change it when we merge

Done.

**Pull request merged**